lib/RoleCredentialResolver.ts

```diff
@@ -0,0 +1,233 @@
+import { RoleCredentialSubject } from '@energyweb/credential-governance';
+import {
+  CredentialResolver,
+  RoleEIP191JWT,
+  RolePayload,
+  upgradeChainId,
+  VerifiableCredential,
+} from '@energyweb/vc-verification';
+import { Resolver } from '@ew-did-registry/did-ethr-resolver';
+import { DidStore } from '@ew-did-registry/did-ipfs-store';
+import {
+  IServiceEndpoint,
+  RegistrySettings,
+} from '@ew-did-registry/did-resolver-interface';
+import { providers, utils } from 'ethers';
+import { CacheServerClient } from './CacheServerClient';
+import * as jwt from 'jsonwebtoken';
+
+export class RoleCredentialResolver implements CredentialResolver {
+  private _ipfsStore: DidStore;
+  private _resolver: Resolver;
+  private _cacheServerClient: CacheServerClient;
+
+  constructor(
+    provider: providers.Provider,
+    registrySetting: RegistrySettings,
+    didStore: DidStore,
+    privateKey: string,
+    cacheServerUrl: string
+  ) {
+    this._ipfsStore = didStore;
+    this._resolver = new Resolver(provider, registrySetting);
+    this._cacheServerClient = new CacheServerClient({
+      privateKey,
+      provider,
+      url: cacheServerUrl,
+    });
+    this._cacheServerClient.login();
```

Comment:
I think there is quite a bit of duplicate from `IpfsCredentialResolver` (https://github.com/energywebfoundation/ew-credentials/blob/develop/packages/vc-verification/src/Ipfs-credential-resolver.ts) here. 

I think we should either accept an IpfsCredentialResolver instance in the constructor and then delegate calls to the IpfsResolver if the cacheServerClient isn't available or construct an instance using the params.